SomosMasApp/SomosMasApp/App/Shared/Modules/Home/Controller/HomeViewController.swift

```diff
@@ -2,30 +2,148 @@
+        collectionView.delegate = self
+        collectionView.contentInset = UIEdgeInsets(top: 0, left: 0, bottom: 0, right: 0)
+        collectionView.register(UINib(nibName: "HomeCollectionViewCell", bundle: nil), forCellWithReuseIdentifier: "mycell")
+        
+        testimoniosCollectionView.isPagingEnabled = true
+        testimoniosCollectionView.dataSource = self
+        testimoniosCollectionView.delegate = self
+        testimoniosCollectionView.contentInset = UIEdgeInsets(top: 0, left: 0, bottom: 0, right: 0)
+        testimoniosCollectionView.register(UINib(nibName: "TestimoniosCollectionViewCell", bundle: nil), forCellWithReuseIdentifier: "Tcell")
+        
+        //lastest news
+        lastestNewsCollectionView.isPagingEnabled = true
+        lastestNewsCollectionView.dataSource = self
+        lastestNewsCollectionView.delegate = self
+        lastestNewsCollectionView.contentInset = UIEdgeInsets(top: 0, left: 0, bottom: 0, right: 0)
+        lastestNewsCollectionView.register(UINib(nibName: "NewsCollectionViewCell", bundle: nil), forCellWithReuseIdentifier: "newscell")
+    }
+    
+}
+extension HomeViewController: UICollectionViewDataSource {
+    
+    func numberOfSections(in collectionView: UICollectionView) -> Int {
+        return 1
+    }
+    
+    func collectionView(_ collectionView: UICollectionView, numberOfItemsInSection section: Int) -> Int {
+        if collectionView == self.collectionView {
+            return sliderData.count
+        } else if collectionView == self.testimoniosCollectionView {
+                return testimoniosData.count
+        } else if collectionView == self.lastestNewsCollectionView {
+            return lastestNewsData.count
+            }
+            return 4
+    }
+    
+    
+    func collectionView(_ collectionView: UICollectionView, cellForItemAt indexPath: IndexPath) -> UICollectionViewCell {
+        if collectionView == self.collectionView {
+            let cell = collectionView.dequeueReusableCell(withReuseIdentifier: "mycell", for: indexPath) as? HomeCollectionViewCell
+            
+            cell?.myTitle.text = sliderData[indexPath.row].title
+            cell?.myImage.image = sliderData[indexPath.row].image
+            cell?.myDescription.text = sliderData[indexPath.row].description
 
-        // Do any additional setup after loading the view.
+            return cell ?? HomeCollectionViewCell()
+        } else if collectionView == self.testimoniosCollectionView {
+            let cell2 = testimoniosCollectionView.dequeueReusableCell(withReuseIdentifier: "Tcell", for: indexPath) as? TestimoniosCollectionViewCell
+            
+            cell2?.testimonioImage.image = testimoniosData[indexPath.row].image
+            cell2?.testimonioEpigrafe.text = testimoniosData[indexPath.row].epigrafe
+            
+            return cell2 ?? TestimoniosCollectionViewCell()
+        } else if collectionView == self.lastestNewsCollectionView { // lastestnews
```

Comment:
never Use if else if else if is very bad coding, you should use a switch or something

Reply:
I code that from another branch that has a if else. Now I pulled the last changes of OT171-39 with switch and rework from there.